src/main/java/io/github/fablabsmc/fablabs/impl/fiber/annotation/AnnotatedSettingsImpl.java

```diff
@@ -319,14 +320,59 @@ private static void checkViolation(Field field) throws FiberException {
 
 		try {
 			field.setAccessible(true);
-			this.applyToNode(sub, field.get(pojo));
+			Object group = field.get(pojo);
+
+			// If group is null, let's try creating a new instance ourselves:
+			if (group == null) {
+				group = tryInstantiateClass(pojo, field.getType());
+				// sets the field so the POJO receives the object we made
+				field.set(pojo, group);
+			}
+
+			this.applyToNode(sub, group);
 			this.applyAnnotationProcessors(pojo, field, sub, this.groupSettingProcessors);
 			sub.build();
 		} catch (IllegalAccessException e) {
 			throw new FiberException("Couldn't fork and apply sub-node", e);
 		}
 	}
 
+	private <P> Object tryInstantiateClass(P pojo, Class<?> clazz) throws FiberException {
+		Constructor<?> constructor = null;
+		boolean accessible = false;
+
+		try {
+			// Inner classes have an extra parameter in their constructor: a reference to the 'upper class'.
+			boolean isInner = !Modifier.isStatic(clazz.getModifiers());
+
+			if (isInner) {
+				constructor = Arrays.stream(clazz.getDeclaredConstructors())
+						.filter(c -> c.getParameterCount() == 1)
+						.findAny()
+						.orElseThrow(() -> new FiberException("Couldn't create group from inner class " + clazz.getSimpleName() + ", is the class public and does it have a nullary constructor?"));
+			} else { // static nested class
```

Comment:
Is there a reason we need to handle inner classes instead of only static nested classes? If so, you can use `Class#getDeclaringClass` to get the exact class of the outer object parameter.

Reply:
Well, optimally I'd like to cover as much use cases as possible (meaning we need to support static and non-static alike), though getting the constructor can definitely be done prettier!